File: Cadastro/att.js
const owner = 'Cavalchi';
const repo = 'Sorteio-Mercedes-Benz';
const path = 'Sorteio-Mercedes-Benz/Cadastro/att.js';
const message = 'mensagem de confirmação';
const token = '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';
const messageElement = document.createElement('div');
document.body.appendChild(messageElement);  

// Obter o conteúdo atual do arquivo
fetch(`https://api.github.com/repos/${owner}/${repo}/contents/${path}`, {
  headers: {
    'Authorization': `token ${token}`
  }
})
.then(response => response.json())
.then(data => {
  // Decodificar o conteúdo do arquivo de base64
  let content = atob(data.content);
  // Adicionar os novos dados do formulário
  content += '\nnovos dados do formulário';
  // Codificar o conteúdo atualizado em base64
  content = btoa(content);
  // Atualizar o conteúdo do arquivo
return fetch(`https://api.github.com/repos/${owner}/${repo}/contents/${path}`, {
  method: 'PUT',
  headers: {
    'Authorization': `token ${token}`,
    'Content-Type': 'application/json'
  },
  body: JSON.stringify({ message, content })
})
.then(response => {
  if (response.ok) {
    // Exibir uma mensagem de sucesso
    messageElement.textContent = 'Dados salvos com sucesso!';
  } else {
    // Exibir uma mensagem de erro
    messageElement.textContent = 'Ocorreu um erro ao salvar os dados.';
  }
})
.catch(error => {
  // Exibir uma mensagem de erro
  messageElement.textContent = 'Ocorreu um erro ao salvar os dados.';
});})

File: Cadastro/att.py
import base64
import json
import os

import requests

owner = 'Cavalchi'
repo = 'Sorteio-Mercedes-Benz'
path = 'Sorteio-Mercedes-Benz/Cadastro/att.js'
message = 'mensagem de confirmação'
token = os.environ['GITHUB_TOKEN']

# Obter o conteúdo atual do arquivo
response = requests.get(
    f'https://api.github.com/repos/{owner}/{repo}/contents/{path}',
    headers={'Authorization': f'token {token}'}
)
data = response.json()

# Decodificar o conteúdo do arquivo de base64
content = base64.b64decode(data['content']).decode('utf-8')

# Coletar os valores dos campos do formulário
email = input('Email: ')
name = input('Nome: ')
lastname = input('Sobrenome: ')
phone = input('Telefone: ')

# Formatar os dados do formulário
form_data = f'{email} {name} {lastname} {phone}'

# Adicionar os novos dados do formulário
content += f'\n{form_data}'

# Codificar o conteúdo atualizado em base64
content = base64.b64encode(content.encode('utf-8')).decode('utf-8')

# Atualizar o conteúdo do arquivo
response = requests.put(
    f'https://api.github.com/repos/{owner}/{repo}/contents/{path}',
    headers={
        'Authorization': f'token {token}',
        'Content-Type': 'application/json'
    },
    data=json.dumps({'message': message, 'content': content})
)

if response.ok:
    # Exibir uma mensagem de sucesso
    print('Dados salvos com sucesso!')
else:
    # Exibir uma mensagem de erro
    print('Ocorreu um erro ao salvar os dados.')
